crates/sp-executor/src/lib.rs

```diff
@@ -102,6 +113,62 @@ impl<Hash: Encode> BundleHeader<Hash> {
     }
 }
 
+// TODO: unify the type between the solution and threshold better.
+pub fn derive_bundle_election_solution(domain_id: DomainId, vrf_output: &[u8]) -> u64 {
```

Comment:
This isn't the right way of doing it, you need something like this (note the APIs used):
https://github.com/subspace/subspace/blob/e18b357254d4e4c766b90f873ede55c318c22502/crates/subspace-verification/src/lib.rs#L291-L306

Reply:
Verification will work properly, but it may not be secure. There was a discussion about this somewhere already, TL;DR is that you should be the code I referenced above to do it correctly, this is also what BABE uses for similar purposes.

Reply:
I'll update it for sure, just wanted to make sure the usage is proper in the code, that's why I asked if there is a test we can add here to ensure that, otherwise it's sort of unwritten knowledge known in a small group of people that using the vrf proof needs extra care.